annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -9,88 +10,226 @@
 import javax.annotation.processing.*;
 import javax.lang.model.SourceVersion;
 import javax.lang.model.element.Element;
+import javax.lang.model.element.PackageElement;
 import javax.lang.model.element.TypeElement;
 import javax.tools.Diagnostic;
+import javax.tools.FileObject;
 import javax.tools.JavaFileObject;
-import java.io.IOException;
-import java.io.PrintWriter;
+import javax.tools.StandardLocation;
+import java.io.*;
+import java.nio.file.Paths;
 import java.util.*;
+import java.util.stream.Collectors;
 
-@SupportedAnnotationTypes("com.exactpro.epfast.annotations.FastType")
+@SupportedAnnotationTypes({"com.exactpro.epfast.annotations.FastType", "com.exactpro.epfast.annotations.FastPackage"})
 @SupportedSourceVersion(SourceVersion.RELEASE_8)
 @AutoService(Processor.class)
 public class FastProcessor extends AbstractProcessor {
+    private static final TypeName SERVICE_TYPENAME = new TypeName("com.exactpro.epfast.ICreator");
+
     private MustacheFactory mustacheFactory = new DefaultMustacheFactory();
 
     private TypeElement fastTypeElement;
 
+    private TypeElement fastPackageElement;
+
+    private HashMap<PackageElement, HashMap<String, Element>> packageElementsMap;
+
     @Override
     public synchronized void init(ProcessingEnvironment processingEnv) {
         super.init(processingEnv);
         fastTypeElement = processingEnv.getElementUtils().getTypeElement(FastType.class.getCanonicalName());
+        fastPackageElement = processingEnv.getElementUtils().getTypeElement(FastPackage.class.getCanonicalName());
     }
 
     @Override
     public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        if (roundEnv.processingOver()) {
+            HashSet<TypeName> creatorImplTypeNames = getNewServiceProviders(packageElementsMap);
+            HashSet<Element> originatingElements = new HashSet<>();
+            packageElementsMap.values().forEach(elements->originatingElements.addAll(elements.values()));
```

Comment:
-> should be surround with whitespace. Add rule for checking this to stylechecler